### ISSUE TITLE
Avoid using id's in css

### ISSUE THREAD

**Issue by katemenkhaus** [vaadin/px-data-grid#239]
Due to the very high specificity of id's in css we don't want them used in css. It is also best to keep js hooks and css hooks separate for maintainability and refactoring.   

Places that use id's in css that should be a class
https://github.com/vaadin/px-data-grid/blob/master/px-data-grid-date-renderer.html#L15
https://github.com/vaadin/px-data-grid/blob/master/px-data-grid-number-renderer.html#L15
https://github.com/vaadin/px-data-grid/blob/master/px-data-grid-selection-column.html#L15
https://github.com/vaadin/px-data-grid/blob/master/px-data-grid-sorter.html#L35
https://github.com/vaadin/px-data-grid/blob/master/px-data-grid-string-renderer.html#L15

**Comment by Toshegg** [comment]
@katemenkhaus, please, take a look at this code:
```html
<input id="editingTemplateInput" class$="{{getClasses(validationResult)}}" type="text" value="{{value::change}}">
      <px-tooltip
        for="editingTemplateInput"
        orientation="top"
        tooltip-message="[[validationResult.message]]"
        opened="[[!validationResult.valid]]"
        ignore-target-events>
```
How to make tooltip to pop up near input without id?
Also, if we are getting rid of id's then wouldn't it be better to use part="foobar" instead of classes?

**Comment by katemenkhaus** [comment]
An id is just a different selector. I would use a class selector instead. 

I'm not very familiar with using parts. I'm curious how you are using it because the spec for it says it isn't ready for implementation.

**Comment by alump** [comment]
@katemenkhaus so even if we would like to keep the ID for JS access, we need to also put class just to use that for styling?

**Comment by alump** [comment]
Fixed in my pull request #249
